pandas/tests/frame/test_reductions.py

```diff
@@ -448,11 +448,15 @@ def test_var_std(self, datetime_frame):
     @pytest.mark.parametrize("meth", ["sem", "var", "std"])
     def test_numeric_only_flag(self, meth):
         # GH 9201
-        df1 = DataFrame(np.random.randn(5, 3), columns=["foo", "bar", "baz"])
+        df1 = DataFrame(np.random.randn(5, 3), columns=["foo", "bar", "baz"]).astype(
+            {"foo": object}
+        )
         # set one entry to a number in str format
         df1.loc[0, "foo"] = "100"
 
-        df2 = DataFrame(np.random.randn(5, 3), columns=["foo", "bar", "baz"])
+        df2 = DataFrame(np.random.randn(5, 3), columns=["foo", "bar", "baz"]).astype(
+            {"foo": object}
+        )
         # set one entry to a non-number str
         df2.loc[0, "foo"] = "a"
```

Comment:
in both `df1` and `df2`, column `'foo'` is initially created to be of dtype `'int'`, and then by setting an element to a string it is upcasting to `object`. This is done on purpose to check the `numeric_only` flag below

Might as well declare column `'foo'` to be of dtype `object` from the start then